README.md

```diff
@@ -84,15 +84,15 @@ Once you have implemented the tools, fill out the report below.
 
 ### Insights you may have had while implementing and comparing the algorithms. 
 
-*Describe this here.*
+It was no problem to implement the algorithms. 
 
 ### Problems encountered if any. 
 
-*Describe this here.*
+I had some problems with the output. I realized that I should have tested better from the beginning. But it worked out in the end.
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+I used the missisipi data and the test data found on Github used to test the algorithms.
```

Comment:
This is not a proper test. One specific string and then random strings only. Random strings, admittedly, tend to catch most errors, but you should always aim at hitting known special cases as well. You don't have to add tests to this project, but you should think about special cases and how to tests them in the future projects.